### PR TITLE
Executable stubs

### PR DESCRIPTION
**Pull request by rkoster**
This PR makes the `load_stub` and `execute_stub` private and in the process fixed a bug in the executable stub implementation.

### CODE REVIEW

File: lib/bosh/workspace/project_deployment.rb [hunk 1]
```diff
@@ -63,14 +54,34 @@ def execute_stub(stub_file)
 
     private
 
+    def stubs_dir
+      File.expand_path(File.join(file_dirname, "../stubs"))
+    end
+
+    def stub_file
+      Dir[File.join(stubs_dir, "#{file_basename('.*')}*")].first
```
Comment (allomov) [inline lib/bosh/workspace/project_deployment.rb]: hey, @rkoster. 
I am not sure about this changes. What if I want to store several stub files in stub folder? What if I use them for different deployments?

Comment (allomov) [inline lib/bosh/workspace/project_deployment.rb]: would you mind if stub file will be determined in the following way:

* first look for option `--stub-file`
* look for file in `stubs` folder with the same name as a deployment name
* and only then take the first file

Comment (rkoster) [inline lib/bosh/workspace/project_deployment.rb]: @allomov I'm fine with adding a `--stub-file` option however `bosh-workspaces` has always been about convention over configuration. With regards to just taking the first, it is not just taking the first. It is taking the first from the glob results which is using the file base name of the deployment file to filter results.

Comment (rkoster) [inline lib/bosh/workspace/project_deployment.rb]: So it is doing `look for file in stubs folder with the same name as a deployment name` but it ignores extensions.

**Pull request merged**